Dec2022Demo/test_S2.jl

```diff
@@ -0,0 +1,134 @@
+using Catlab.CategoricalAlgebra
+using Catlab.Present, Catlab.Theories
+using AlgebraicPetri
+using AlgebraicPetri: Graph
+
+include("LibForOct2022Demo.jl");
+
+# Type system model
+types′ = LabelledPetriNet([:Pop],
+    :infect=>((:Pop, :Pop)=>(:Pop, :Pop)),
+    :disease=>(:Pop=>:Pop),
+    :strata=>(:Pop=>:Pop))
+types = map(types′, Name=name->nothing)
+
+# Parts of type system for ease of reference
+s, = parts(types′, :S)
+t_interact, t_disease, t_strata = parts(types′, :T)
+i_interact1, i_interact2, i_disease, i_strata = parts(types′, :I)
+o_interact1, o_interact2, o_disease, o_strata = parts(types′, :O);
+
+# SIRD augmented model
+SIRD_aug = LabelledPetriNet([:S, :I, :R, :D],
+	  :inf => ((:S, :I)=>(:I, :I)),
+	  :rec => (:I=>:R),
+	  :death => (:I=>:D),
+	  :id => (:S => :S),
+	  :id => (:I => :I),
+	  :id => (:R => :R)
+	)
+AlgebraicPetri.Graph(SIRD_aug)
+
+# Typed SIRD model
+SIRD_aug_typed = ACSetTransformation(SIRD_aug, types,
+S = [s, s, s, s],
+T = [t_interact, t_disease, t_disease, t_strata, t_strata, t_strata],
+I = [i_interact1, i_interact2, i_disease, i_disease, i_strata, i_strata, i_strata],
+O = [o_interact1, o_interact2, o_disease, o_disease, o_strata, o_strata, o_strata],
+Name = name -> nothing # specify the mapping for the loose ACSet transform
+)
+@assert is_natural(SIRD_aug_typed)
+AlgebraicPetri.Graph(dom(SIRD_aug_typed))
+
+num_ages = 2
+
+# Function to assemble a multi-region model
+function makeMultiAge(n,f_aug=false)
```

Comment:
Kris wrote a function for this: https://github.com/AlgebraicJulia/ASKEM.jl/blob/32099bb026625a24ca061cde7da22ee5ac754879/src/Stratify.jl#L65

Reply:
Also, we don't use camelCase in julia for function names :)